test/run-tests.py

```diff
@@ -597,15 +641,30 @@ def RunTest(info, options, variables, verbose_level=0):
   os.makedirs(out_dir)
   variables['out_dir'] = out_dir
 
-  cmd = info.GetCommand(rel_gen_input_path, variables, options.arg,
-                        verbose_level)
-  if options.print_cmd:
-    print(' '.join(cmd))
+  total_stdout = ''
+  total_stderr = ''
+  returncode = 0
+  total_duration = 0
 
-  try:
-    return RunCommandWithTimeout(cmd, cwd, timeout, verbose_level > 0, env)
-  except (Error, KeyboardInterrupt) as e:
-    return e
+  for cmd in info.cmds:
+    cmd_list = cmd.AsList(variables, options.arg, verbose_level)
```

Comment:
I don't really like the name `AsList` but I can't think of a better one right now.  Something like ".Get()" or ".Format()"?  I don't really like those either :)